ezsystems/ezplatform-http-cache
# EZP-29194: Prefix tags with repository id to support multi repo on same proxy

> Improvment: https://jira.ez.no/browse/EZP-29194

Varnish/Fastly is able to keep page cache separated based on host name and Url, however tags
are currenly not seperated. So to avoid cache claring by tags affecting
several installs which is using different databases, we need to prefix
them using repository name _(if set)_.

Review notes:
- Had to set fos cache instance as alias to make sure what we have configured as arguments are taken into account. Also since the class have been the same since 0.5, this does not change much besides cleanup the compiler pass.
- I opted to decorate Purge Client to try to avoid implementations having to be adapted, including Fastly bundle.
- I opted to skip prefix, if no repository is set, this is the default when installing eZ Platform

#### QA
Can be tested by setting up ezplatform.yml in following way:
```yml
    repositories:
        intranet:
            storage:
(...)
    system:
        site_group:
            repository: intranet
```

With that make sure to test that cache is cleared as you would normally expect, and if you want to see under the hood on tags, make sure to configure your ip as debug ip in varnish so you see headers.
